kong/pdk/request.lua

```diff
@@ -370,18 +371,49 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the normalized path component of the request's URL. The return
+  -- value is the same as `kong.request.get_raw_path()` but normalized according
+  -- to RFC 3986 section 6:
+  --
+  -- * Percent-encoded values of unreserved characters are decoded (`%20`
+  --   becomes ` `).
+  -- * Percent-encoded values of reserved characters have their hexidecimal
+  --   value uppercased (`%2f` becomes `%2F`).
+  -- * Relative path elements (`/.` and `/..`) are dereferenced.
+  -- * Duplicate slashes are consolidated (`//` becomes `/`).
+  --
+  -- @function kong.request.get_path
+  -- @phases rewrite, access, header_filter, response, body_filter, log, admin_api
+  -- @tparam[opt=true] boolean merge_slashes consolidate duplicate slashes in the path
+  -- @treturn string the path
+  -- @usage
+  -- -- Given a request to https://example.com/t/Abc%20123%C3%B8%2f/parent/..//test/./
+  --
+  -- kong.request.get_path() -- "/t/Abc 123ø%2F/test/"
+  function _REQUEST.get_path()
+    return normalize(_REQUEST.get_raw_path(), true)
```

Comment:
This would change the default behavior of this API, any possibility to add another API like `get_normalized_path`?

Reply:
This is actually by design. We want `get_path()` to be normalized so that it's secure by default when used for string comparison. Sorry I didn't make this more apparent in the description! This reminds me to add a breaking change entry for the changelog though.

Reply:
It is a security fix, so the change is expected.